성현/algorithms/src/main/java/chapter2/RemoveDuplicates.java

```diff
@@ -0,0 +1,52 @@
+package chapter2;
+
+import java.util.*;
+
+public class RemoveDuplicates {
+
+    public boolean checkValid(List<Integer> intArr) {
+        return intArr.size() == new HashSet(intArr).size();
+    }
+
+    public List<Integer> removeDuplicates(List<Integer> input) {
+        Map<Integer, Boolean> intMap = new HashMap<>();
+        List<Integer> answer = new ArrayList<>();
+
+        for (Integer i : input) {
+            if (intMap.getOrDefault(i, false)) {
+                continue;
+            }
+            intMap.put(i, true);
+            answer.add(i);
+        }
+
+        return answer;
+    }
+
+    public List<Integer> removeDuplicatesWithoutBuffer(int[] input) {
```

Comment:
배열인자 하나라면 가변인자를 써도 좋을 것 같네요

```suggestion
    public List<Integer> removeDuplicatesWithoutBuffer(int... input) {
```

Reply:
오홍 좋은 제안 감사합니다